README.md

```diff
@@ -1,119 +1,231 @@
 Cartography
 ===========
 
+[![NPM version][npm-image]][npm-url] [![Dependency Status][daviddm-url]][daviddm-image]
 [![Build Status](https://secure.travis-ci.org/Adslot/cartography.png?branch=master)](http://travis-ci.org/Adslot/cartography)
 
 Javascript Object to Object mapper
 
 Cartography takes an input Object and translates it into a new Object with a
-different structure according to a schema.
+different structure as specified by a schema.
 It can also be used for validating input structures.
 
-```coffeescript
-{same, from, map, filters, CartographyError} = require 'cartography'
-{required, isInteger, isOneOf} = filter
-
-isPositive = (v) -> if v > 0 then v else throw new CartographyError 'must be positive'
-
-cartographyUserSchema =
-  id: same isInteger
-  username: from 'email', required, (s) -> s.toLowerCase()
-  password: same required, isString
-  provider: -> 'Adslot'
-  maxShare: from 'sales.salePercentage', parseInt, isInteger, isPositive
-  address:
-    streetName: from 'addressStreet', required, isString
-    streetNumber: from 'addressNumber', required, isString
-    state: from 'addressState', isOneOf ['QLD', 'SA', 'ACT', 'NSW', 'VIC']
-
-userInDBFormat = map userInTransferFormat, cartographyUserSchema
+
+```javascript
+
+  var theInput = {
+    id: 123,
+    userName: 'HappyLand',
+    color: {
+      definition: '#7ff'
+    },
+    day: 'Mon',
+    month: 'Aug',
+  }
+
+  var theDesiredOutput = {
+    id: 123,
+    name: 'HappyLand',
+    base: {
+      color: '#7FF',
+      time: 'Mon - Aug'
+    }
+  }
+
+  // enters Cartography
+  var f = cartography.filters
+
+  var theSchemaToTranslateOneIntoTheOther = {
+    id: [f.required, f.isInteger],
+    name: ['userName', f.isString],
+    base: {
+      color: ['color.definition', f.isString, function(v) {return v.toUpperCase()}],
+      time: function(inputObject) {return inputObject.day +' - '+ inputObject.month}
+    }
+  }
+
+  assert.deepEqual(cartography.map(theInput, theSchemaToTranslateOneIntoTheOther), theDesiredOutput)
 ```
-(All examples are in CoffeeScript, but Cartography is pure JavaScript.)
 
-The schema's structure resembles that of the output Object: it has the
+The schema's structure is the same as that of the output Object: it has the
 same attribute names and the same nested structure (if any), but each
```

Comment:
This isn't true if some of the mappers returned `undefined`, right? Not sure if that's worth clarifying